content/3_Silver/Prefix_Sums_2.mdx

```diff
@@ -439,6 +439,56 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class ForestQueries {
+	static int N;
+	static int Q;
+	static int[][] pfx;
+	static int[][] arr;
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+
+		N = io.nextInt();
+		Q = io.nextInt();
+
+		pfx = new int[N+1][N+1];
+		arr = new int[N+1][N+1];
+
+		for(int i = 0; i < N; i++) {
+			String line = io.next();
+			for(int j = 0; j < N; j++) {
+				if(line.charAt(j) == '*') {
+					arr[i+1][j+1]++;
+				}
+			}
+		}
+
+		for(int i = 1; i < N + 1; i++) {
+			for(int j = 1; j < N + 1; j++) {
+				pfx[i][j] = arr[i][j] + pfx[i-1][j] + pfx[i][j-1] - pfx[i-1][j-1];
+			}
+		}
+
+		for(int i = 0; i < Q; i++) {
+			int x1 = io.nextInt();
+			int y1 = io.nextInt();
+			int x2 = io.nextInt();
+			int y2 = io.nextInt();
+			io.println(pfx[x2][y2] - pfx[x1-1][y2] - pfx[x2][y1-1] + pfx[x1-1][y1-1]);
+		}
+		io.close();
+	}
+	CodeSnip{Kattio}
```

Comment:
```suggestion

	CodeSnip{Kattio}
```

Looks nicer with extra space.

Also, this doesn't compile... Kattio needs to be declared as "static"

Reply:
However that will be a tomorrow problem (unless sm1 else takes care of it) cuz I gon go to sleep